hazelcast/src/main/java/com/hazelcast/map/impl/query/PartitionScanExecutor.java

```diff
@@ -28,4 +30,25 @@
 public interface PartitionScanExecutor {
 
     Collection<QueryableEntry> execute(String mapName, Predicate predicate, Collection<Integer> partitions);
+
+    /**
+     * Executes the predicate on a partition chunk. The offset in the partition is defined by the {@code tableIndex}
+     * and the soft limit is defined by the {@code fetchSize}. The method returns the matched entries and an
+     * index from which new entries can be fetched which allows for efficient iteration of query results.
+     * <p>
+     * <b>NOTE</b>
+     * The reading should be done when the {@link IMap} is static. When reading from a Map, with or without
+     * predicate or projection, it is possible the map may be being mutated. It is also possible that
+     * partitions will be migrated to admit new members or adjusting after a member has left. It is possible to
+     * miss reading entries or read an entry twice.
+     *
+     * @param mapName     the map name
+     * @param predicate   the predicate which the entries must match
+     * @param partitionId the partition which is queried
+     * @param tableIndex  the index from which entries are queried
+     * @param fetchSize   the soft limit for the number of entries to fetch
+     * @return entries matching the predicate and a table index from which new entries can be fetched
+     */
+    QueryableEntriesWithCursor executeOnPartitionChunk(String mapName, Predicate predicate, int partitionId, int tableIndex,
+                                                       int fetchSize);
```

Comment:
Then this would be called 
`QueryableEntriesWithCursor execute(String mapName, Predicate predicate, int partitionId, int tableIndex, int fetchSize);`